modmail/bot.py

```diff
@@ -19,27 +23,130 @@ class ModmailBot(commands.Bot):
 
-    async def create_session(self) -> None:
-        """Create an aiohttp client session."""
-        self.http_session = ClientSession()
+    @classmethod
+    def create(cls, *args, **kwargs) -> "ModmailBot":
+        """
+        Create a ModmailBot instance.
 
-    async def start(self, *args, **kwargs) -> None:
+        This configures our instance with all of our custom options, without making __init__ unusable.
+        """
+        intents = Intents(
+            guilds=True,
+            messages=True,
+            reactions=True,
+            typing=True,
+            members=True,
+            emojis_and_stickers=True,
+        )
+        # start with an invisible status while we load everything
+        status = discord.Status.invisible
+        activity = Activity(type=discord.ActivityType.listening, name="users dming me!")
+        # listen to messages mentioning the bot or matching the prefix
+        # ! NOTE: This needs to use the configuration system to get the prefix from the db once it exists.
+        prefix = commands.when_mentioned_or(CONFIG.bot.prefix)
+        description = "Modmail bot by discord-modmail."
+        # allow commands not caring of the case
+        case_insensitive_commands = True
+        # allow only user mentions by default.
+        # ! NOTE: This may change in the future to allow roles as well
+        allowed_mentions = AllowedMentions(everyone=False, users=True, roles=False, replied_user=True)
+        return cls(
+            case_insensitive=case_insensitive_commands,
+            description=description,
+            status=status,
+            activity=activity,
+            allowed_mentions=allowed_mentions,
+            command_prefix=prefix,
+            intents=intents,
+            **kwargs,
+        )
+
+    def run(self, *args, **kwargs) -> None:
         """
-        Start the bot.
 
-        This just serves to create the http session.
+        Start up our instance of the bot. Since this method is blocking, it must be called last.
+
+        This method sets up the bot, loads extensions and plugins, and then executes the main task.
+
+        A blocking call that abstracts away the event loop
+        initialisation from you.
+        If you want more control over the event loop then this
+        function should not be used. Use :meth:`start` coroutine
+        or :meth:`connect` + :meth:`login`.
+        Roughly Equivalent to: ::
+            try:
+                loop.run_until_complete(start(*args, **kwargs))
+            except KeyboardInterrupt:
+                loop.run_until_complete(close())
+                # cancel all tasks lingering
+            finally:
+                loop.close()
+        .. warning::
+            This function must be the last function to call due to the fact that it
+            is blocking. That means that registration of events or anything being
+            called after this function call will not execute until it returns.
```

Comment:
This is wrong? start() is not an alternative for this method of startup, also, this isn't our language 👀